### ISSUE TITLE
Dataset in Echarts 4 does not work properly with line series that have the areaStyle prop.

### ISSUE THREAD

**Issue by lzp4ever** [ecomfe/echarts#7536]
### One-line summary [问题简述]
Dataset in Echarts 4 does not work properly with line series that have the `areaStyle` prop.

### Version & Environment [版本及环境]
+ ECharts version [ECharts 版本]: 4.0.2
+ Browser version [浏览器类型和版本]: chrome 65
+ OS Version [操作系统类型和版本]: arch linux

### Expected behaviour [期望结果]

With areaStyle prop, line series should have filled color from x axis to the line.
But now the base line turn to the line segment between the first data's coordinate to the last.

### ECharts option [ECharts配置项]

```javascript
option = {
  legend: {},
  tooltip: {
    trigger: 'axis',
  },
  dataset: {
    source: [
      ['order', '1', '2', '3', '4', '5', '6'],
      ['a', 41.1, 30.4, 65.1, 53.3, 83.8, 98.7],
      ['b', 86.5, 92.1, 85.7, 83.1, 73.4, 55.1],
      ['c', 24.1, 67.2, 79.5, 86.4, 65.2, 82.5],
      ['d', 55.2, 67.1, 69.2, 72.4, 53.9, 39.1],
    ],
  },
  xAxis: { type: 'category' },
  yAxis: { gridIndex: 0 },
  series: [
    { type: 'line', smooth: true, seriesLayoutBy: 'row', areaStyle: {} },
    { type: 'line', smooth: true, seriesLayoutBy: 'row', areaStyle: {} },
    { type: 'line', smooth: true, seriesLayoutBy: 'row', areaStyle: {} },
    { type: 'line', smooth: true, seriesLayoutBy: 'row', areaStyle: {} },
  ],
};
```

### Other comments [其他信息]

JSFiddle: [http://jsfiddle.net/cj31rtpf/36/](http://jsfiddle.net/cj31rtpf/36/)

**Comment by wu007kun** [comment]
![image](https://user-images.githubusercontent.com/26088782/36015687-8324ff16-0daa-11e8-8699-cf55cf3eb20f.png)

**Comment by 100pah** [comment]
Fixed on bf0df08
